### PR TITLE
Consume WebJobs diagnostics for Application Insights flushing issues

### PR DESCRIPTION
**Pull request by michaelpeng36**
### Issue describing the changes in this PR

Consumes diagnostic events raised by the WebJobs SDK that signal when there are flushing issues in the ApplicationInsightsLoggerProvider. See this [PR](https://github.com/Azure/azure-webjobs-sdk/pull/2939) for the relevant changes in the WebJobs SDK.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] My changes **do not** require diagnostic events changes
    * [ ] Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by liliankasem — APPROVED** [review]

File: src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs [hunk 1]
```diff
@@ -63,7 +65,7 @@ private void SubscribeListeners()
                             _listenerSubscriptions.Add(listener.Name, listenerSubscription);
                         }
 
-                        _logger.LogInformation("Subscribed to host diagnostic source '{sourceName}'", listener.Name);
+                        _logger.LogInformation("Subscribed to diagnostic source '{sourceName}'", listener.Name);
```
Comment (liliankasem) [inline src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs:68]: Can you double check that changing this log won't impact any detectors?

**Review by RohitRanjanMS — APPROVED** [review]

**Comment by liliankasem** [comment]
@michaelpeng36 any updates on this PR? Is it ready to merge?

**Comment by liliankasem** [comment]
Blocked: Pending WebJobs changes and release

File: src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs (hunk 1 above)
Comment (michaelpeng36) [reply src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs:68]: Thanks for the reminder, Lilian! I just took a look through the detectors repository, and from what I see, there does not appear to be a detector that searches explicitly for the string `"Subscribed to host diagnostic..."`, so I think this change shouldn't break anything on that front.

Comment (liliankasem) [reply src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs:68]: Thanks for checking!

**Comment by liliankasem** [comment]
@michaelpeng36 is this still blocked?

**Comment by michaelpeng36** [comment]
@liliankasem the PR should be unblocked now, as the WebJobs SDK has been released and merge conflicts resolved. Thanks for your patience!

File: NuGet.config [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <add key="azure_app_service_staging" value="https://www.myget.org/F/azure-appservice-staging/api/v2" />
     <add key="azure-appservice-test" value="https://azfunc.pkgs.visualstudio.com/e6a70c92-4128-439f-8012-382fe78d6396/_packaging/azure-appservice-test%40Local/nuget/v3/index.json" />
     <add key="Microsoft.Azure.Functions.PowerShellWorker" value="https://azfunc.pkgs.visualstudio.com/e6a70c92-4128-439f-8012-382fe78d6396/_packaging/Microsoft.Azure.Functions.PowerShellWorker/nuget/v3/index.json" />
+    <add key="AzureFunctions@staging" value="https://azfunc.pkgs.visualstudio.com/e6a70c92-4128-439f-8012-382fe78d6396/_packaging/AzureFunctions%40staging/nuget/v3/index.json" />
```
Comment (liliankasem) [inline NuGet.config:9]: Do we still need this staging feed?

**Review by liliankasem — APPROVED** [review]

File: NuGet.config (hunk 2 above)
Comment (michaelpeng36) [reply NuGet.config:9]: Yes, this is part of the effort to move package feeds to ADO that @kashimiz is driving. We're still including the staging feed to catch issues found in staging packages earlier.

**Pull request merged**